src/net.c

```diff
@@ -3207,27 +3162,126 @@ coap_join_mcast_group_intf(coap_context_t *ctx, const char *group_name,
 
   if (result != 0) {
     coap_log(LOG_ERR,
-             "coap_join_mcast_group: cannot resolve multicast address: %s\n",
-             gai_strerror(result));
+             "coap_join_mcast_group_intf: %s: "
+              "Cannot resolve multicast address: %s\n",
+             group_name, gai_strerror(result));
     goto finish;
   }
 
+/* Need to do a windows equivalent at some point */
+#ifndef _WIN32
+  if (ifname) {
+    /* interface specified - check have correct IPv4 / IPv6 information */
+    int done_ip4 = 0;
+    int done_ip6 = 0;
+    int ip4fd;
+    struct ifreq ifr;
+
+    /* See which mcast address family types are being asked for */
+    for (ainfo = resmulti; ainfo != NULL; ainfo = ainfo->ai_next) {
```

Comment:
Maybe leave the loop if `(done_ip4 == 1 && done_ip6 == 1)`?

Reply:
Have added it in, even though group address is likely to be a single literal multicast  IP.